style/react/README.md

```diff
@@ -0,0 +1,50 @@
+# React
+
+ * Use React in [Strict Mode]
+ * Use React with [TypeScript](../typescript/README.md)
+ * Use [Jest](../testing-jest/README.md) for unit testing
+ * Use [styled-components] for CSS-in-JS
+ * Use [Apollo] with [code generation] when working with [GraphQL]
+ * Use [React Router] for routing in SPAs
+ * Avoid deeply nested routing when using React Router
+ * Prefer [Function Components] over [Class Components]
+ * Prefer keeping a single component in each file
+ * Use `PascalCase` for component names and their file names
+ * Use [React Hooks]
+ * Use pre-built hooks when possible (e.g. [streamich/react-use])
+ * Use [custom hooks] to encapsulate stateful logic outside a component
+ * Avoid [Forwarding Refs]
```

Comment:
Is this saying to prefer passing a ref as a custom prop like `inputRef` instead of using `forwardRef` or to avoid passing refs altogether?

Reply:
My thought here is probably better summarized as `Avoid nesting Forward Refs`.

The guideline I'd like to include here is less about how you should pass a ref, and more around when their use might point to an overly complex design.

I have seen some cases where there are multiple levels of components forwarding a ref, and it becomes increasingly difficult to trace the ref from its assignment to its usage. Furthermore a ref forwarded multiple layers and used by some far-off component creates a coupling between the two components and also pollutes the component tree when debugging with the React dev tools.

These have just been my observations, curious to hear others' thoughts too.